etc/docs/generate-docs.ts

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env ts-node
+
+import { parse, stringify } from '@iarna/toml';
+import * as child_process from 'child_process';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import { promisify } from 'util';
+
+const exec = promisify(child_process.exec);
+
+const RELEASES_TOML_FILE = './template/data/releases.toml';
+const RELEASES_JSON_FILE = './template/static/versions.json';
+
+interface JsonVersionSchema {
+  version: string;
+}
+
+interface VersionSchema {
+  version: string;
+  status: 'current' | 'supported' | 'not-supported';
+  api: string;
+  usesMongoDBManual?: boolean;
+  docs?: string;
+  semvarVersion: string;
+}
+
+interface TomlVersionSchema {
+  current: string;
+  mongodDBManual: string;
+  versions: VersionSchema[]
+}
+
+/**
+ * ATTENTION - EDIT THIS BEFORE RELEASING
+ */
+const NEW_VERSION: VersionSchema = {
+  // version - the display name for the version of the driver on the docs website
+  //  ex: Driver 4.5
+  version: 'Driver 4.5',
+
+  // status - a status for the version.  typically would probably just be 'latest'
+  status: 'current',
+
+  // api - the name of the folder for the generated documentation to live.  typically should be version_id prefixed with `./`
+  //  ex: ./4.5
+  api: './4.5',
+
+  // usesMongoDBManual - if true, includes a link to the mongodb documentation for the new version
+  usesMongoDBManual: true,
+
+  // the 
+  semvarVersion: '4.5'
+};
+
+function validateVersionInformation(jsonVersions: JsonVersionSchema[]) {
+  const isVersionInfoValid = ['version', 'version_id', 'status', 'api'].every(
+    key => NEW_VERSION[key] !== ''
+  );
+
+  if (!isVersionInfoValid) {
+    console.error(
+      'Error - version information invalid.  Please update the `NEW_VERSION` object before running the script.'
+    );
+    process.exit(1);
+  }
+
+  if (jsonVersions.some(({ version }) => version === NEW_VERSION.semvarVersion)) {
+    console.error(
+      'Error - attempting to publish docs for a release that already exists.'
+    );
+    process.exit(1);
+  }
+
+}
+
+async function copyNewDocsToGeneratedSite() {
+  const outputDirectory = `./temp/${NEW_VERSION.semvarVersion}`;
+  const pathToBuiltDocs = './build';
+  const command = `cp -R ${pathToBuiltDocs} ${outputDirectory}`;
+  return await exec(command);
+}
+
+async function updateSiteTemplateForNewVersion(tomlData: TomlVersionSchema, jsonData: JsonVersionSchema[]) {
+  tomlData.versions.unshift(NEW_VERSION);
+  tomlData.current = NEW_VERSION.version;
+  writeFileSync(RELEASES_TOML_FILE, stringify(tomlData as any));
+
+  jsonData.unshift({ version: NEW_VERSION.semvarVersion });
```

Comment:
Should this allow for updating the docs that were already generated for a given version? I think at least for main/"next", it would be nice to regenerate and update docs continuously instead of just the one time

Reply:
It could - I'm happy to update the script to support that if we wanted.  FYI though, although the script supports "main/next" simply because you can name your version whatever you like, it would take a bit more effort to make it work well.  If that's something we want to support in this PR, I can also make those changes

Reply:
Just curious - what would be the issues we'd need to work around with using something like `Next` instead of a real version? Mainly I'm just concerned with always being able to properly preview the current docs and to generate them in a consistent spot so that they *can* be included in PRs

Reply:
I assume we'd always want `main` at the top of the list of driver versions.  So basically just add logic to detect when `main` is passed as the version, and make sure to place it at the top of the list driver versions on the generated static site

Reply:
I have a suggestion in a new comment to handle this with a sort, let me know what you think